torch/_dynamo/variables/builder.py

```diff
@@ -514,12 +515,20 @@ def wrap_tensor(self, value: torch.Tensor):
                         source=None,
                         # Guards are added inside register_attr_or_module
                     )
+
+                # For example_value, this input could be an intermediate tensor
+                # from a graph break, where this tensor has requires_grad field
+                # set to True. If we just pass on the value, it would seem like
+                # a leaf variable for this subgraph, and can unnecessary raise
+                # assertions like "inplace mutation of leaf variable".
+                # To workaround, we clone the tensor for the example_value, which
+                # sets the requires_grad field to True only if its not a leaf.
                 tensor_variable = TensorVariable.create(
                     tx=self.tx,
                     proxy=self.tx.output.create_graph_input(
                         re.sub(r"[^a-zA-Z0-9]+", "_", self.name), type(value)
                     ),
-                    example_value=value,
+                    example_value=clone_input(value),
```

Comment:
Is value a real tensor or a fake tensor? From the rest of the module here, it looks real. Wouldn't cloning it be bad; e.g., shouldn't we fakeify it? I don't know what the `example_value` invariants are though.

Reply:
This is a real tensor as of now. This is coming direcly from the python frame. We fakify it later.

Reply:
But there are real failures in CI, so let me take a look :|

Reply:
if the tensor is really large, cloning it will aid OOM. And then you'll have to revisit this clone when you do memory optimization work. is there options to fakify it around here...

Reply:
> if the tensor is really large, cloning it will aid OOM. And then you'll have to revisit this clone when you do memory optimization work. is there options to fakify it around here...

This is the input tensor only and not an intermediate tensor. Intermediate tensors are always recorded as fake tensor. But, yes, it could be problematic if the input tensor itself is really large, or there are a lot of graph breaks.

There are some places in the code where we use this real value, through `real_value_cache`. So, I cannot turn this into fake tensor right away. Let me open an issue to track this separately.

~~~
torch/_dynamo/output_graph.py:        self.real_value_cache = {}
torch/_dynamo/output_graph.py:                self.real_value_cache.pop(node, None)
torch/_dynamo/output_graph.py:            self.real_value_cache.clear()
torch/_dynamo/output_graph.py:                self.real_value_cache.pop(node, None)
torch/_dynamo/output_graph.py:        self.real_value_cache.clear()
torch/_dynamo/variables/tensor.py:    cache = output_graph.real_value_cache
torch/_dynamo/variables/tensor.py:                proxy.tracer.real_value_cache[proxy.node] = example_value
~~~

Reply:
Added - https://github.com/pytorch/torchdynamo/issues/1751

Reply:
`real_value_cache` is only used in export though right ? Could we avoid filling it otherwise ?